src/android/CameraPreviewFragment.java

```diff
@@ -551,20 +627,33 @@ public void setUpCamera(JSONObject options, ProcessCameraProvider cameraProvider
                     .build();
         }
 
-        Size targetResolution = null;
-        if (targetSize > 0) {
-            targetResolution = CameraPreviewFragment.calculateResolution(getContext(), targetSize);
-        }
+        targetResolution = calculateResolution(getContext(), targetSize, aspectRatio);
```

Comment:
```suggestion
        this.targetResolution = calculateResolution(getContext(), targetSize, aspectRatio);
```

Should refactor everywhere